Update Docs Core Concepts pages

ZYJLiu opened the pull request:
Update Core Concepts section of docs

Include following pages:
- Solana Account Model
- Transactions and Instructions
- Program on Solana
- Program Derived Address
- Cross Program Invocations
- Tokens on Solana

jacobcreech commented on docs/core/accounts.md:
```diff
@@ -1,177 +1,143 @@
+
+By default, all new accounts are owned by the
+[System Program](https://github.com/solana-labs/solana/tree/master/programs/system/src).
+The System Program is one of Solana's "Native Programs", which comes "built-in"
```
You introduce "Native Programs" without defining it. Recommend defining the different programs you can have before going into the System Program. Could be a small section right before this.

jacobcreech replied:
These are also known as "Native accounts", so that may be a nice way to start a definition and segue into the System Program.

ZYJLiu replied:
updated with intro and link to https://docs.solanalabs.com/runtime/programs

jacobcreech commented on docs/core/accounts.md:
```diff
@@ -1,177 +1,143 @@
+   account data as defined in the program logic
+
+This data account creation process is often abstracted as a single step, but
+it's helpful to understand the underlying process.
```
Could we get nice conclusions to summarize the learnings on this page? Highlight the most important bits at the very least.

ZYJLiu replied:
Not sure if it makes sense to add summary at end of each page for docs since it's not likely going to be read from top to bottom. Maybe keypoint bullet points at top of page?

jacobcreech replied:
They're docs. Are they not supposed to be read from top to bottom?

We can do the keypoint bullets at the top. That's what we did with cookbook. We also did summaries to drive in the points.

ZYJLiu replied:
What I meant is that readers might not necessarily always reach the bottom of the page (for example if they're just scanning through quickly for a specific topic)

ZYJLiu replied:
added key points to top of pages

jacobcreech reviewed (commented):
Initial review. Lots more to cover on my end

jacobcreech commented on docs/core/transactions/retry.md:
```diff
@@ -1,321 +0,0 @@
----
```
This page is super important. Is there a way we can keep this?

ZYJLiu replied:
added back under Advanced Concepts

jacobcreech commented on docs/core/transactions/fees.md:
```diff
@@ -1,130 +0,0 @@
----
```
While I agree that fees are confusing and scattered throughout the docs, is there a single place you'd put them? I would avoid deleting this page until the _one page to rule them all_ is created

ZYJLiu replied:
added back under Advanced Concepts

jacobcreech reviewed (commented):
Mid review again. Looking good so far. Could we add summaries on the bottom of each page?

jacobcreech commented on docs/core/cpi.md:
```diff
@@ -1,360 +1,100 @@
-For seed schemes that are susceptible to this type of hash collision, a common
-remedy is to insert separators between seeds, e.g. transforming {"abc", "def"}
-into {"abc", "-", "def"}.
+### CPI with PDA Signer
```
Could you go into more of why you would want a signer here? https://solana.com/docs/core/cpi#instructions-that-require-privileges goes pretty in depth into it, especially with the common use case of a program owning a token account

jacobcreech commented on docs/core/tokens.md:
```diff
@@ -0,0 +1,504 @@
+For reference, here is a Solana Explorer link to the USDC
+[Mint Account](https://explorer.solana.com/address/EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v).
+
+### Create Mint Account
```
Can we do this earlier as "Create a Token", explain what you did following "hey, you actually created a mint account", and do with CLI on Solana Playground/local instead?

jacobcreech replied:
Goal is to show the devs what possible, then explain it. Get them their first developer success asap.

ZYJLiu replied:
Including both js and cli examples on the docs right now might require creating separate pages because there is not a component for creating tabs yet, and the way the page is parsed right now if you have multiple h tags with the same name it will always navigate to the first one. 

In the example below, clicking CLI or JS on Example 2, will always navigate to Example 1
```
## Example 1
### CLI
### JS

## Example 2
### CLI
### JS
```

Although for the tokens page on docs, I think it might make sense to provide context before examples since some readers might just be looking for an explanation instead of running examples.

jacobcreech replied:
What I'm saying is prioritize using CLI only right now for tokens. Let's leave adding JS for when our code component supports multiple languages.

ZYJLiu replied:
Spent some time figuring out how to create token with the CLI. I think the JS examples would be more beneficial since they provide more context on how to interact with tokens, and act as additional references for how to build transactions/instructions. Each example already has a corresponding Solana Playground link where a reader can just press "run" to run the example. Plus, there are generally more question on Stack Exchange related to JS than how to use the CLI.

With the CLI (even on Solana Playground), it would require telling the reader to copy paste various outputs of one CLI command into the next one.

ZYJLiu replied:
Initially, I had the Tokens page immediately following the Transaction and Instructions page to provide more examples of building transactions and a practical example of the Solana account model.

ZYJLiu replied:
added cli examples

jacobcreech commented on docs/core/tokens.md:
````diff
@@ -0,0 +1,504 @@
+First, generate a keypair to use as the address of the new account and calculate
+the amount of lamports needed for the Token Account data.
+
+```ts
````
Similar, CLI please!

jacobcreech commented on docs/core/tokens.md:
```diff
@@ -0,0 +1,504 @@
+sender can include instructions to create the recipient's token account, which
+generally is the Associated Token Account.
+
+## Token Metadata
```
Let's actually go through the motions of doing the metadata setup with CLI. Complete the full concept from start to finish in one page.

ZYJLiu replied:
added cli example, although using token extensions flags requires a local install of the cli bc its not supported on playground yet.

its looks like playground cli is [custom](https://github.com/solana-playground/solana-playground/blob/0573c957e54f30f844b786660a9c5c6f22874d0e/wasm/spl-token-cli/src/cli.rs) and mirrors the functionality of spl-token cli

jacobcreech reviewed (commented):
Got through the review. Really good read, learned some things. LMK if any of the comments are confusing

jacobcreech commented:
Let's clean these up and get what we have live, then move on to the other pages

ZYJLiu commented on docs/core/transactions.md:
````diff
@@ -1,235 +1,332 @@
+instructions, along with the signature for the transaction.
+
+```
+"transaction": {
````
do you happen to know what [`indexToProgramIds`](https://github.com/solana-labs/solana-web3.js/blob/master/packages/library-legacy/src/message/legacy.ts#L66) and [`stackHeight`](https://github.com/solana-labs/solana/blob/master/sdk/src/inner_instruction.rs) are for? 

it seems like stack_height has to do with cpis, but nothing shows up if its just 1 cpi. does it have to be a nested cpi?

also not sure what kind of transaction would result in `indexToProgramIds` to show up in the logs

jacobcreech replied:
Not sure, were you still looking for an answer on this?

ZYJLiu replied:
still not sure, asked on stackexchange
added explanation for other parts of the transaction logs

nickfrosty replied:
I vote we add a tracking issue for this and add it later when we have all the info

ZYJLiu commented on docs/core/transactions.md:
````diff
@@ -1,235 +1,332 @@
-allows multiple transactions to repeat the exact same action. Transactions also
-have lifetimes that are defined by the blockhash, as any transaction whose
-blockhash is too old will be rejected.
+```typescript
````
put basic example first, after intro

ZYJLiu commented on docs/core/transactions.md:
```diff
@@ -1,235 +1,332 @@
+client libraries. However, if one is not available, you can always fall back to
+manually building the instruction.
+
+## Compact-Array Format
```
updated to put as part of transaction section, was combining content from cookbook and wasn't sure where to put it

ZYJLiu commented on docs/core/pda.md:
````diff
@@ -0,0 +1,428 @@
+[Rust](https://github.com/solana-labs/solana/blob/master/sdk/program/src/pubkey.rs#L484)),
+but in this section, we will walk through examples using Javascript.
+
+```ts
````
removed implementation snippets, and just linked to source

ZYJLiu commented on docs/core/cpi/example2.md:
````diff
@@ -0,0 +1,263 @@
+);
+```
+
+### 1. Anchor Format
````
rename to Anchor CpiContext?

jacobcreech replied:
Maybe just "How to CPI with signers using Anchor?"

ZYJLiu commented on docs/core/tokens.md:
```diff
@@ -0,0 +1,504 @@
+---
+
+This section will cover the basics of how tokens are represented on Solana. You
+can think of tokens as a counter in a database that also keeps track of the
```
rewrote

ZYJLiu commented on docs/core/transactions.md:
```diff
@@ -1,235 +1,332 @@
+The total size of the signatures and message for a transaction is limited to
+1232 bytes.
+
+![Transaction Format](/assets/docs/core/transactions/issues_with_legacy_txs.png)
```
added explaination, images reused from cookbook

ZYJLiu commented on docs/core/transactions.md:
```diff
@@ -1,235 +1,332 @@
+  "keys": [
+    {
+      "pubkey": "3z9vL1zjN6qyAFHhHQdWYRTFAcy69pJydkZmSFBKHg1R",
+      "isSigner": true,
```
the explaination is included as part of the linked example

ZYJLiu replied:
added context to linked sol transfer example

ZYJLiu commented on docs/core/programs.md:
```diff
@@ -1,92 +1,24 @@
-  "A Solana Program, aka smart contract, is the executable code that interprets
-  the instructions on the blockchain. There are two types: Native and on chain."
+title: Programs
+sidebarLabel: Programs on Solana
```
wym just added link to the "developing programs" section, which has a various pages on details related to programs

ZYJLiu replied:
added section of verifiable builds and bpf, native programs are mentioned as part of solana account model page

ZYJLiu commented on docs/core/pda.md:
```diff
@@ -0,0 +1,428 @@
+A PDA can then be used as the address (unique identifier) for an on-chain
+account, providing a method to easily store, map, and fetch program state.
+
+![Off Curve Address](/assets/docs/core/pda/address-off-curve.svg)
```
remade images

jacobcreech commented:
Ready for another round of reviews @ZYJLiu ?

ZYJLiu commented:
yep ready 👍

nickfrosty commented on docs/index.md:
```diff
@@ -1,5 +1,5 @@
 ---
-sidebarSortOrder: 0
+sidebarSortOrder: 1000
```
why are you trying to put the docs index at the bottom?

ZYJLiu replied:
based on vercel data, it looks like currently the most viewed routes under docs are:

/cluster
/wallet
/rpc
/terminology

rpc and terminology are probably so high up because its the first thing on the top of the page.
it doesn't really make sense for these pages to be so high up as they're primariliy references


eventually, I was thinking of redoing the Intro/Quickstart to starts with an overview of how to use Solana Playground

nickfrosty replied:
We will adjust this one later with the updates in the intro section that jacob is working on (and the secondary nav). But for now, let's keep it as.

- RPC is one of the most searched sections since those pages are well indexed and linked on the internet
- the terms page is heavily trafficed since it is linked all over, so as people browse the docs or guides, they often click through to get the definition of a given word
- wallet - this one is a mystery.... we suspect high search intent and domain authority. e.g. "solana wallet" sort of makes sense to show a page on solana-com that is about wallets

nickfrosty reviewed (commented):
My review of this PR is still in progress, but here are my current comments. 

- why did you move the pages that were under core/transactions to the advanced section?
- I think retrying and versioned makes sense to do this move, but not fees
- in fact, I think this core/transactions/fees and the core/rent doc should be combined into a generic "core/fees" docs
  - with a rework / merging with the intro pages (or better yet deleting those intro pages)
- why did you delete the transaction confirmation page?
- why did you delete the rent doc?
- why did you delete the runtime doc?
- it looks like you split the cpi doc into 3 different docs? and added some other info into them?
  - why split into multiple vs add all the content into the original cpi doc?

throughout, for any of the pages your moved or deleted, you did not update the links to them in the docs you did not modify. these links all need to be updated to their correct final location

jacobcreech commented on docs/core/tokens.md:
```diff
@@ -0,0 +1,564 @@
+- A Mint Account represents a unique token on the network and stores global
+  metadata such as total supply.
+
+- A Token Account tracks individual ownership of tokens for a specific mint
+  account.
+
+- An Associated Token Account is a Token Account created with an address derived
+  from the owner's and mint account's addresses.
```
There's something called an ancillary token account. Token accounts can be either associated or ancillary.

nickfrosty replied:
oooh fun fact! I had not heard that term before. but I guess it makes sense

ZYJLiu commented:
- mainly trying to clean up core concepts section so its just the most basics topics to get started, the addition transaction pages are kind of more "good to know" type of info. Confirmation/Retry are also pretty big pages that I haven't looked into updating. Using Advanced Concepts section to put pages that have topics to keep, but pages not updated.
- there's two transaction fee pages for some reason which probably needs to get consolidated. Also version transaction and LUT should be updated as one page since LUT is the only use case for versioned transactions.
- agree those should get combined
- will add this back under advanced topic, must of missed when adding back
- don't think rent really needs its own page, its mentioned briefly under Solana Account Model page
- deleted bc its more of a page with random facts and not really an overview of the runtime, which would be more like [joe's notion page on the topic](https://fluff-ranunculus-275.notion.site/The-Agave-Runtime-d1f8d3608e5d4529b120e09e80b48887)
- split the cpi doc into three bc of the issue with the table of contents where you can't have multiple headings with the same name (it always navs to the first one).

any recommendations for best way to find the various links that need to be updated?

> My review of this PR is still in progress, but here are my current comments.
> 
> * why did you move the pages that were under core/transactions to the advanced section?
> * I think retrying and versioned makes sense to do this move, but not fees
> * in fact, I think this core/transactions/fees and the core/rent doc should be combined into a generic "core/fees" docs
>   
>   * with a rework / merging with the intro pages (or better yet deleting those intro pages)
> * why did you delete the transaction confirmation page?
> * why did you delete the rent doc?
> * why did you delete the runtime doc?
> * it looks like you split the cpi doc into 3 different docs? and added some other info into them?
>   
>   * why split into multiple vs add all the content into the original cpi doc?
> 
> throughout, for any of the pages your moved or deleted, you did not update the links to them in the docs you did not modify. these links all need to be updated to their correct final location

ZYJLiu commented on docs/core/transactions.md:
```diff
@@ -1,235 +1,408 @@
-state. Instructions invoke programs that make calls to the Solana runtime to
-update the state (for example, calling the token program to transfer tokens from
-your account to another account).
+- The maximum size of a transaction is 1232 bytes.
```
added to page content

ZYJLiu commented on docs/core/transactions.md:
```diff
@@ -1,235 +1,408 @@
-[instructions](/docs/terminology.md#instruction), the runtime will process each
-of the instructions contained within the transaction, in order, and atomically.
-If any part of an instruction fails, then the entire transaction will fail.
+## Key Points
 
```
thats how I initially had it, but the docs pages doesn't have a way to scroll to top (above the first h tag) so readers would manually have to scroll to navigate back to an "intro paragraph"

ZYJLiu replied:
swaped overview and keypoints

nickfrosty replied:
thanks for the swap. I am working on a PR for solana-com and will add some sort of "scroll to top" type thing in the sidebar

ZYJLiu commented on docs/core/accounts.md:
```diff
@@ -1,177 +1,186 @@
-rent-exempt before the increase will remain rent-exempt afterwards
+## Key Points
+
+- Accounts can store up to 10MB of data, which can consist of either executable
```
added

ZYJLiu commented on docs/core/accounts.md:
```diff
@@ -1,177 +1,186 @@
+- **Buffer Account**: A temporary account that stores byte code while a program
+  is being actively deployed or upgraded. Once the process is complete, the data
+  is transferred to the Program Executable Data Account and the buffer account
+  is closed.
```
yea buffer accounts are [automatically closed](https://github.com/solana-labs/solana/blob/27eff8408b7223bb3c4ab70523f8a8dca3ca6645/programs/bpf_loader/src/lib.rs#L583-L590) (unless the deployment fails)

ZYJLiu commented on docs/core/programs.md:
```diff
@@ -1,92 +1,94 @@
 
-### Native programs
+- [Native Rust](/docs/core/programs/rust): This approach involves writing Solana
+  programs directly in Rust without leveraging any frameworks. It offers more
```
under the hood, anchor crate also use the solana crates, just gonna remove word directly

ZYJLiu commented on docs/core/programs/rust.md:
```diff
@@ -0,0 +1,546 @@
+---
+title: Native Rust
+sidebarLabel: Native Rust
```
the "developing programs" section has a bunch of misc content and also not sure if its all correct / up to date.

prefer to keep these basic examples as part of core concepts so reader has context of basic program structure while scanning through core concepts (vs getting directed to guides or separate section)

nickfrosty commented:
> * mainly trying to clean up core concepts section so its just the most basics topics to get started, the addition transaction pages are kind of more "good to know" type of info. Confirmation/Retry are also pretty big pages that I haven't looked into updating. Using Advanced Concepts section to put pages that have topics to keep, but pages not updated.

gotcha, makes sense. I'm game! FYI: I just updated confirmation a few days ago, which is also why it is also a merge conflict on your PR

> * there's two transaction fee pages for some reason which probably needs to get consolidated. Also version transaction and LUT should be updated as one page since LUT is the only use case for versioned transactions.

totally agree on the version and LUt page being combined (and most of the versions doc going into a guide). there is a task item in the notion doc that has this item as well

> * agree those should get combined

nice! great minds :)

> * will add this back under advanced topic, must of missed when adding back

sounds good

> * don't think rent really needs its own page, its mentioned briefly under Solana Account Model page

let's combine the contents of this with the transaction fees page, making one master core concepts "fees" page

> * deleted bc its more of a page with random facts and not really an overview of the runtime, which would be more like [joe's notion page on the topic](https://fluff-ranunculus-275.notion.site/The-Agave-Runtime-d1f8d3608e5d4529b120e09e80b48887)

while I agree about it being full of "random facts", it also has most of the docs information about the compute budget. so we need to at least move that content somewhere that makes sense. maybe inside that "master fees" core concept doc I mentioned above? compute and compute budget is sort of a fee? the doc could be titled "fees and compute" or something similar?

> * split the cpi doc into three bc of the issue with the table of contents where you can't have multiple headings with the same name (it always navs to the first one).

that makes more sense as to why split them. I think we should put the sub pages into guides or the new cookbook section. and maybe add more context into the main cpi doc if needed. this also aligns with your first point above about simplifying the core concepts section

> any recommendations for best way to find the various links that need to be updated?

the best option is to just `ctrl+f` the project in your editor

ZYJLiu commented on docs/core/tokens.md:
````diff
@@ -0,0 +1,564 @@
+
+Signature: 44fvKfT1ezBUwdzrCys3fvCdFxbLMnNvBstds76QZyE6cXag5NupBprSXwxPTzzjrC3cA6nvUZaLFTvmcKyzxrm1
+```
+
````
adding a callout to Token Examples section

ZYJLiu commented on docs/core/tokens.md:
```diff
@@ -0,0 +1,564 @@
+The Token Extensions Program enables additional metadata (such as name, symbol,
+link to image) to be stored directly on the Mint Account.
+
+<Callout>
+   Using the Token Extensions CLI flags requires a local installation of the CLI:
+   
+   `cargo install --version 3.4.0 spl-token-cli`
+</Callout>
```
reworded

ZYJLiu commented on docs/core/pda.md:
```diff
@@ -0,0 +1,386 @@
+same `user` address as a seed, then the transaction will fail. This is because
+an account will already exist at the derived address.
+
+## How to derive a PDA
```
swapped

ZYJLiu commented on docs/core/pda.md:
```diff
@@ -0,0 +1,386 @@
+- **Optional seeds**: Predefined inputs (ex. string, number, other account
+  addresses) used to derive a PDA. These inputs are converted to a buffer of
+  bytes.
+- **Bump seed**: An additional input (with a value between 255-0) that is used
```
this is mentioned under "CreateProgramAddress" section

the "Bump seed" bullet point is already pretty long, but adding that it expliclity starts with 255

ZYJLiu commented on docs/core/pda.md:
```diff
@@ -0,0 +1,386 @@
+
+### Canonical Bump
+
+The first bump seed (255-0) that derives a valid PDA is referred to as the
```
this is mentioned under "CreateProgramAddress" section

ZYJLiu commented on docs/core/cpi.md:
````diff
@@ -1,360 +1,122 @@
-client.send_and_confirm_message(&[&alice_keypair, &bob_keypair], &message);
-```
+- The callee program can make additional CPIs to other programs, up to a maximum
+  depth of 4.
````
added a callout to [max_invoke_stack_height](https://github.com/solana-labs/solana/blob/27eff8408b7223bb3c4ab70523f8a8dca3ca6645/program-runtime/src/compute_budget.rs#L138), which seems like the hardcoded constant of 5.  (1 for initial call, and limit to 4 cpi depth) 

https://github.com/solana-labs/solana/blob/27eff8408b7223bb3c4ab70523f8a8dca3ca6645/program-runtime/src/compute_budget.rs#L31-L35

nickfrosty replied:
amazing! thanks!

ZYJLiu commented on docs/core/cpi.md:
```diff
@@ -1,360 +1,122 @@
+  maximum depth of 4 (ex. B->C, C->D)
+- The programs can "sign" on behalf of the [PDAs](/docs/core/pda) derived from
+  its program ID
 
```
added callout

ZYJLiu commented on docs/core/cpi.md:
```diff
@@ -1,360 +1,122 @@
+The
+[`invoke`](https://github.com/solana-labs/solana/blob/27eff8408b7223bb3c4ab70523f8a8dca3ca6645/sdk/program/src/program.rs#L132)
+function is used when making a CPI that does not require PDA signers. When
+making CPIs, signers provided to the caller program can extend to the callee
```
yea, both pda and non-pda signers

ZYJLiu commented on docs/core/cpi/how-to-cpi.md:
````diff
@@ -0,0 +1,238 @@
+}
+```
+
+The `test.ts` file demonstrates how to invoke the custom `sol_transfer`
````
doesn't look like it

ZYJLiu commented on docs/core/cpi/how-to-cpi.md:
```diff
@@ -0,0 +1,238 @@
+        },
+    );
+
+    transfer(cpi_context, amount)?;
```
yea its an anchor helper, added snippet

ZYJLiu commented on docs/core/programs/rust.md:
````diff
@@ -0,0 +1,546 @@
+let signer = next_account_info(accounts_iter)?;
+let system_program = next_account_info(accounts_iter)?;
+```
+
````
with rust iterators, you have to manually get the next item in the iterator. `next_account_info` is the helper to get the accountinfo for the next account in the iterator

ZYJLiu commented on docs/core/programs/rust.md:
```diff
@@ -0,0 +1,546 @@
+
+## Client
+
+Interacting with Solana programs written in native Rust involves directly
```
added callout

ZYJLiu commented on docs/core/programs/anchor.md:
```diff
@@ -0,0 +1,739 @@
+
+The main macros found in an Anchor program include:
+
+- [`declare_id`](/docs/core/programs#declare_id): Specifies the program's
+  on-chain address
+- [`#[program]`](/docs/core/programs#program): Specifies the module containing
+  the program’s instruction logic
+- [`#[derive(Accounts)]`](/docs/core/programs#derive-accounts): Applied to
+  structs to indicate a list of accounts required for an instruction
+- [`#[account]`](/docs/core/programs#account): Applied to structs to create
+  custom account types specific to the program
```
fixed

ZYJLiu commented on docs/core/programs/anchor.md:
```diff
@@ -0,0 +1,739 @@
+
+Use `memcmp` to filter for accounts storing data that matches a specific value
+at a specific offset. When calculating the offset, remember that the first 8
+bytes are reserved for the account discriminator. Using `memcmp` requires you to
```
reword

ZYJLiu commented on docs/core/transactions.md:
```diff
@@ -1,235 +1,408 @@
 
-[Programs](/docs/core/programs.md) on Solana don’t store data/state; rather,
-data/state is stored in accounts.
+### Overview
```
removed all the overview tags and moved to top of page

ZYJLiu commented on docs/core/transactions.md:
```diff
@@ -1,235 +1,408 @@
+- **is_writable**: Specify if the account data will be modified
+
+This information is referred to as the
+[AccountMeta](https://github.com/solana-labs/solana/blob/27eff8408b7223bb3c4ab70523f8a8dca3ca6645/sdk/program/src/instruction.rs#L539).
```
updated image

nickfrosty commented on docs/core/cpi/how-to-cpi-with-signer.md:
```diff
@@ -0,0 +1,272 @@
+[`system_instruction::transfer`](https://github.com/solana-labs/solana/blob/27eff8408b7223bb3c4ab70523f8a8dca3ca6645/sdk/program/src/system_instruction.rs#L881)
+to build the instruction.
+
+First, add these imports to the top of `lib.rs`:
```
nice!

nickfrosty commented on docs/core/programs/anchor.md:
````diff
@@ -0,0 +1,740 @@
+```
+
+When integrating with a frontend using
+the[wallet adapter](https://solana.com/developers/guides/wallets/add-solana-wallet-adapter-to-nextjs),
````
```suggestion
the [wallet adapter](https://solana.com/developers/guides/wallets/add-solana-wallet-adapter-to-nextjs),
```

nickfrosty commented on docs/core/programs/rust.md:
```diff
@@ -0,0 +1,555 @@
+Similarly, fetching and deserializing account data requires creating a schema
+compatible with the on-chain program's data structures.
+
+<Callout>
+  There are multiple client languages supported. You can find details for [Rust](docs/clients/rust) and [Javascript/Typescript](docs/clients/rust) under the Solana Clients of the documentation.
+</Callout>
```
the links are broken:

```suggestion
<Callout>
  There are multiple client languages supported. You can find details for [Rust](/docs/clients/rust) and [Javascript/Typescript](/docs/clients/rust) under the Solana Clients of the documentation.
</Callout>
```

nickfrosty commented on docs/intro/dev.md:
```diff
@@ -145,8 +145,7 @@ Transactions fees are calculated based on two main parts:
 
-You can read more about transaction fees
-[here](/docs/core/transactions/fees.md).
+You can read more about transaction fees [here](/docs/intro/transaction_fees).
```
this link is invalid. it should be `/docs/intro/transaction-fees`

ZYJLiu replied:
change to `/docs/core/fees` for the updated fee page

nickfrosty commented on docs/intro/economics.md:
```diff
@@ -43,7 +43,7 @@ issuance over time. Next is a brief section on
 and how token dilution might influence staking behavior.
 
-An overview of [Transaction Fees](/docs/core/transactions/fees.md) on Solana is
+An overview of [Transaction Fees](/docs/intro/transaction_fees) on Solana is
```
this link is invalid. it should be `/docs/intro/transaction-fees`

ZYJLiu replied:
change to `/docs/core/fees` for the updated fee page

nickfrosty commented on docs/programs/index.md:
```diff
@@ -61,7 +61,7 @@ manner.
 sent to write all of the buffered byte-code to the program's data account. This
 either mark the new program as
-[`executable`](/docs/core/accounts.md#executable), or complete the process to
+[`executable`](/docs/terminology#onchain-program), or complete the process to
```
this seems like a useless link to change too. I vote to just remove the link

ZYJLiu replied:
removed

nickfrosty commented on docs/terminology.md:
```diff
@@ -273,7 +273,7 @@ header, array of account addresses, recent [blockhash](#blockhash), and an array
 
 Learn more about the
-[message formatting inside of transactions](/docs/core/transactions.md#message-format)
+[message formatting inside of transactions](/docs/core/transactions#transaction)
```
linking to the actual message header makes more sense

ZYJLiu replied:
updated

ZYJLiu commented:
moved cpi/program examples out to guides as discussed
